File: go/clase1/ciudades/ciudades.go
package main
import (
	"fmt"
	"sort"
	ut "module/utilidades"
	"strconv"
	"strings"
)

func main() {
	const nroCiudades int =6 
	var (
		ciudades [] string 
		ciudad string 
	)

	ut.Encabezado("Ingreso de ciudades", 50)
	for i := 0; i < nroCiudades; i++ {
		ciudad = ut.LeerTexto("Ingrese la ciudad " + strconv.Itoa(i+1) )
		ciudades = append(ciudades, ciudad)
	}

	verCiudades(ciudades, "listado de ciudades", 50)

	sort.Strings(ciudades)
	verCiudades(ciudades, "ciudades ordenadas ascendentemente", 50)
	
	sort.Sort(sort.Reverse(sort.StringSlice(ciudades)))
	verCiudades(ciudades, "ciudades ordenadas descendentemente: ", 50)
	
}

func verCiudades ( ciudades []string, titulo string, largo int)  {
	ut.Encabezado(titulo, largo)
	for _, item := range ciudades {
		fmt.Println("La ciudad es: ", strings.ToUpper(item))
	}
}

File: java/project2/src/arsistema/Ciudades.java
package arsistema;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

public class Ciudades {
    public static void main(String[] args) {
        // Declarar variables
        List<String> ciudades = new ArrayList<>();
        // String ciudad = new String(); forma vieja
        String ciudad = "";
        final int nroCiudades = 5;

        Util.Encabezado("ingreso de ciudades:", 50);
        for (int i = 0; i < nroCiudades; i++) {
            ciudad = Util.leerTexto("Ingresar la ciudad " + (i + 1), "ciudad");
            ciudades.add(i, ciudad);
        }

        
        verCiudades(ciudades, "ver ciudades", 50);
        ciudades.sort(Comparator.naturalOrder());
        verCiudades(ciudades,"Ver ciudades ordendas ascendentemente:", 50);
        ciudades.sort(Comparator.reverseOrder());
        verCiudades(ciudades, "Ver ciudades ordendas ascendentemente:", 50);
       
    }
    public static void verCiudades(List<String> ciudades, String titulo, int largo){
        Util.Encabezado(titulo, largo);
        ciudades
        .stream()
        .forEach((item) -> System.out.println("La ciudad es " + item.toUpperCase()));
    }
}
